### ISSUE TITLE
missing status text "Not Found" in response status line

### ISSUE THREAD

**Issue by joekyo** [actix/actix-web#490]
```
extern crate actix_web;
use actix_web::{server, App, HttpResponse};

fn main() {
    server::new(|| {
        App::new()
        .prefix("/app")
        .resource("", |r| r.f(|_| HttpResponse::Ok()))  // <- handle `/app` path
        .resource("/", |r| r.f(|_| HttpResponse::Ok())) // <- handle `/app/` path
        .resource("/test", |r| {
            r.get().f(|_| HttpResponse::Ok());
            r.head().f(|_| HttpResponse::MethodNotAllowed());
        })
    }).bind("localhost:3000")
        .unwrap()
        .run();
}
```
```
$ curl -I http://localhost:3000/app/z
HTTP/1.1 404 Not Found  // <= it works here

$ curl -I http://localhost:3000/app
HTTP/1.1 200 OK

$ curl -I http://localhost:3000/ap
HTTP/1.1 404  // <= missing `Not Found` here
```

**Comment by DoumanAsh** [comment]
Hm, this is quite interesting find, so we have status code, but not its text representation in response?

**Comment by Neopallium** [comment]
If no one else is working on this, I will have a look and create a testcase for it later today.

**Comment by fafhrd91** [comment]
Great! Just assign ticket to yourself

**Comment by Neopallium** [comment]
I can't assign myself to this ticket.

**Comment by Neopallium** [comment]
fixed #498
